### PR TITLE
Fix/ok response

### PR DESCRIPTION
**Pull request by veptechno**
Add status codes and error information (if necessary) to the response for the chat adapter. And return a 404 error if no channel is registered in the WebhookConnector.

### CODE REVIEW

File: channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/polling/ResponseSender.kt [hunk 1]
```diff
@@ -9,15 +12,23 @@ import io.ktor.http.*
...
         try {
-            val serializedResponse: JaicpBotResponse = response.await() ?: return
-            client.post<String>("$url/sendMessage".toUrl()) {
-                body = serializedResponse
-                contentType(ContentType.parse("application/json"))
+            val response = deferredResponse.await()
+            when (response) {
+                is JaicpBotResponse -> send(url, response)
+                is JaicpErrorResponse -> logger.warn("Failed to process request, message ${response.message}")
+                is JaicpAsyncResponse -> logger.debug("Ignored an async response")
```
Comment (Denire) [inline channels/jaicp/src/main/kotlin/com/justai/jaicf/channel/jaicp/polling/ResponseSender.kt:21]: I think log for JaicpAsyncReponse should either be trace or dont exist at all.

File: core/src/main/kotlin/com/justai/jaicf/channel/http/HttpBotResponse.kt [hunk 2]
```diff
@@ -12,19 +12,61 @@ import java.nio.charset.StandardCharsets
  */
 data class HttpBotResponse(
     val output: ByteArrayOutputStream,
-    val contentType: String
+    val contentType: ContentType,
+    val statusCode: HttpStatusCode = HttpStatusCode.OK
```
Comment (Denire) [inline core/src/main/kotlin/com/justai/jaicf/channel/http/HttpBotResponse.kt:16]: I'd suggest making statusCode a simple Integer. There is no point in limiting codes to our enum.

File: core/src/main/kotlin/com/justai/jaicf/channel/http/Ktor.kt [hunk 3]
```diff
@@ -43,14 +44,14 @@ fun Routing.httpBotRouting(vararg channels: Pair<String, HttpBotChannel>) {
...
-            when (response) {
-                null -> call.respond(HttpStatusCode.NotFound, "Bot didn't respond")
-                else -> call.respondOutputStream(ContentType.parse(response.contentType)) {
-                    response.output.writeTo(this)
-                }
+            call.respondOutputStream(
+                ContentType.parse(response.contentType.value),
+                HttpStatusCode.fromValue(response.statusCode.value)
+            ) {
+                response.output.writeTo(this)
```
Comment (Denire) [inline core/src/main/kotlin/com/justai/jaicf/channel/http/Ktor.kt:53]: Could you please check if this blocking code causes any errors on ktor threads? Shouldn't we move it to another coroutine context?

File: examples/hello-world/src/main/kotlin/com/justai/jaicf/examples/helloworld/channel/JaicpPoller.kt [hunk 4]
```diff
@@ -25,9 +25,8 @@ fun main() {
                 AimyboxChannel,
                 AlexaChannel,
                 ActionsFulfillmentDialogflow()
-            )
+            ),
+            url = "http://localhost:9020"
```
Comment (Denire) [inline examples/hello-world/src/main/kotlin/com/justai/jaicf/examples/helloworld/channel/JaicpPoller.kt:29]: You forgot to delete it.

**Review by Denire — CHANGES_REQUESTED** [review]

File: core/src/main/kotlin/com/justai/jaicf/channel/http/Ktor.kt (hunk 3 above)
Comment (veptechno) [reply core/src/main/kotlin/com/justai/jaicf/channel/http/Ktor.kt:53]: Ktor itself runs it in the IO context.

**Review by Denire — APPROVED** [review]

**Review by morfeusys — APPROVED** [review]

**Pull request merged**